core/src/com/unciv/logic/map/TileMap.kt

```diff
@@ -139,22 +139,31 @@ class TileMap {
         unit.assignOwner(civInfo, false)
 
         var unitToPlaceTile : TileInfo? = null
-        // try to place at the original point (this is the most probable scenario)
-        val currentTile = get(position)
-        if (unit.movement.canMoveTo(currentTile)) unitToPlaceTile = currentTile
-
-        // if it's not suitable, try to find another tile nearby
-        if (unitToPlaceTile == null) {
+        // try to place at the original point (this is the most probable scenario), then move through passable tiles outwards
+        // this is done using a set, initializing it with the given position, testing whether any tile in the set is
+        // suitable (unit can move to), and if not expanding the set with all neighbors the unit could pass through, repeat.
+        // The loop is potentially done twice for land units, the first time allowing only land tiles
+        // and if this fails a second pass allows embarkment
+        val potentialCandidates = mutableSetOf<TileInfo>()
+        val isLand = unit.baseUnit.unitType.isLandUnit()
+        var allowEmbark = unit.baseUnit.unitType.isWaterUnit()
+        val isOther = !(isLand || allowEmbark)      // air and ..? to be safe, no domain check
+        while (true){
+            potentialCandidates.clear()
+            potentialCandidates.add(get(position))
```

Comment:
The initial position check can be moved out of the loop and checked before it, no need to check it twice.